tests/elliptic_curves/util.py

```diff
@@ -1,31 +1,335 @@
+    for ix, el in enumerate(elements[::-1]):
+        out += nums_to_script([el])
+        if ix != len(elements) - 1:
+            out += Script.parse_string("OP_EQUALVERIFY")
+        else:
+            out += Script.parse_string("OP_EQUAL")
+    return out
+
+
+def generate_verify_point(P, degree: int) -> Script:  # noqa: N803
+    """Generate verification script for P.
+
+    Args:
+        - `P`: The point for which to generate the verification script.
+        - `degree` (int): The extension degree of the curve E for which P belongs to E. Each coordinate
+        of P is made of `degree` elements.
+
+    """
     out = Script()
-    if not point_p.is_infinity():
-        for ix, el in enumerate(point_p.to_list()[::-1]):
-            out += nums_to_script([el])
-            if ix != len(point_p.to_list()) - 1:
-                out += Script.parse_string("OP_EQUALVERIFY")
-            else:
-                out += Script.parse_string("OP_EQUAL")
+    if not P.is_infinity():
+        out += generate_verify_from_list(P.to_list())
     else:
         out += Script.parse_string(
-            "0x00 OP_EQUALVERIFY 0x00 OP_EQUALVERIFY " * (degree - 1) + "0x00 OP_EQUALVERIFY 0x00 " "OP_EQUAL"
+            "0x00 OP_EQUALVERIFY 0x00 OP_EQUALVERIFY " * (degree - 1) + "0x00 OP_EQUALVERIFY 0x00 OP_EQUAL"
         )
     return out
 
 
-def generate_unlock(point_p, degree) -> Script:
+def generate_extended_list(elements: List[List[int]], positions_elements: List[int], filler: int = 1) -> List[int]:
+    """Take a list of element a fills it with the filler element.
+
+    Args:
+        - elements (List[List[int]]): the list of elements (each being a list itself) to be extended.
+        - positions_elements (List[int]): the positions that the elements should take in the extended list.
+        - filler (int = 1): the filler element.
+
+    Example:
+    >>> generate_extended_list([1,2,3],[3,1,0],10)
+    [1,10,2,3]
+
+    >>> generate_extended_list([1,2,3],[3,2,0],10)
+    [1,2,10,3]
+
+    >>> generate_extended_list([1,2,3],[3,2,1],10)
+    [1,2,3,10]
+
+    >>> generate_extended_list([[1,2],3,4],[4,2,1],10)
+    [1,2,3,4,10]
+
+    >>> generate_extended_list([[1,2],3,4],[5,2,1],10)
+    [1,2,10,3,4,10]
+
+    >>> generate_extended_list([1,2,3],[10,4,2],15)
+    [1,15,15,15,15,15,2,15,3,15,15]
+    """
+    if type(positions_elements) is dict:
+        positions_elements = positions_elements.values()
+    extended_list = []
+    positions_elements = [*positions_elements, -1]
+
+    for i in range(len(elements)):
+        element_to_add = elements[i]
+        extended_list = [*extended_list, *element_to_add]
+        extended_list = [
+            *extended_list,
+            *[filler] * (positions_elements[i] - positions_elements[i + 1] - len(element_to_add)),
+        ]
+
+    return extended_list
+
+
+def generate_test(
+    modulus: int,
+    P,  # noqa: N803
+    Q,  # noqa: N803
+    positions: Dict[str, int],
+    negations: Dict[str, int],
```

Comment:
Type of negations should be Dict[str, bool] isn't it?

Reply:
(same for rolls)

Reply:
Thanks